code/_onclick/item_attack.dm

```diff
@@ -24,7 +24,7 @@
 	return FALSE
 
 /obj/attackby(obj/item/I, mob/living/user, params)
-	return ..() || (can_be_hit && I.attack_obj(src, user))
+	return ..() || (can_be_hit && istype(I) && I.attack_obj(src, user))
```

Comment:
This isn't the proper way to fix things; you're papering over the problem. This is the "I can't see it runtiming, so it's not a problem" solution.

You need to fix this at the source; why is it runtiming to begin with? How and why is a non-item getting passed to `attackby`? Fix it at the source.